src/core/Form/Select/BaseSelect/SelectItem/SelectItem.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { Component, ReactNode, MouseEvent } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { HtmlLi } from '../../../../../reset';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../../../theme';
+import { Icon } from '../../../../Icon/Icon';
+import { baseStyles } from './SelectItem.baseStyles';
+
+const baseClassName = 'fi-select-item';
+
+const selectItemClassNames = {
+  wrapper: `${baseClassName}_wrapper`,
+  hasKeyboardFocus: `${baseClassName}--hasKeyboardFocus`,
+  queryHighlight: `${baseClassName}--query_highlight`,
+  selected: `${baseClassName}--selected`,
+  disabled: `${baseClassName}--disabled`,
+  icon: `${baseClassName}_icon`,
+};
+
+export interface SelectItemProps {
+  /** SelectItem container div class name for custom styling. */
+  className?: string;
+  /** Indicates if the current item has keyboard focus. */
+  hasKeyboardFocus: boolean;
+  /** Query for matching string type item children for highlighting text. */
+  hightlightQuery: string | undefined;
+  checked: boolean;
+  disabled?: boolean;
+  id?: string;
+  onClick: (event: MouseEvent<HTMLLIElement>) => void;
+  children: ReactNode;
```

Comment:
Maybe we could add JSDoc for these props.

Reply:
This was already discussed and agreed that maybe it's not necessary to document props that have self explaining names like the ones here. More complex ones already have docs.